manifests/config/context/resourcelink.pp

```diff
@@ -0,0 +1,81 @@
+# Definition: tomcat::config::server::resourcelink
+#
+# Configure ResourceLink elements in $CATALINA_BASE/conf/context.xml
+#
+# Parameters:
+# - $catalina_base is the base directory for the Tomcat installation.
+# - $resource_ensure specifies whether you are trying to add or remove the
+#   ResourceLink element. Valid values are 'true', 'false', 'present', and
+#   'absent'. Defaults to 'present'.
+# - $global The name of the linked global resource in the global JNDI context.
+# - $resource_type The fully qualified Java class name expected by the web application 
+#   when it performs a lookup for this resource link.
+# - $resource_link_name The name of the resource link to be created, relative 
+#   to the java:comp/env context.
+# - An optional hash of $additional_attributes to add to the Connector. Should
+#   be of the format 'attribute' => 'value'.
+# - An optional array of $attributes_to_remove from the Connector.
+define tomcat::config::context::resourcelink (
+  $resource_link_name    = undef,
+  $global                = undef,
+  $resource_type         = undef,
+  $catalina_base         = $::tomcat::catalina_home,
+  $resource_ensure       = 'present',
+  $additional_attributes = {},
+  $attributes_to_remove  = [],
+) {
+  if versioncmp($::augeasversion, '1.0.0') < 0 {
+    fail('Server configurations require Augeas >= 1.0.0')
+  }
+
+  validate_re($resource_ensure, '^(present|absent|true|false)$')
+  
+  if $resource_link_name {
+    $_resource_link_name = $resource_link_name
+  } else {
+    $_resource_link_name = $name
+  }
+
+  $base_path = "Context/ResourceLink[#attribute/name='${_resource_link_name}']"
+
+  if $resource_ensure =~ /^(absent|false)$/ {
+    $changes = "rm ${base_path}"
+  } else {
```

Comment:
again, curious to see if this will be easier on the eye once we've replaced all the if/else below with [selectors](http://docs.puppetlabs.com/puppet/3.7/reference/lang_conditional.html#selectors)